File: aula11/fatorial.py
# coding: utf-8

def fatorial(n):
	if n < 0:
		raise ValueError("Não existe fatorial de n<0")
	elif n <= 1:
		return 1
	else:
		return n * fatorial(n-1)

numero = input("Digite um número inteiro positivo:")
print str(numero)+"! = ", fatorial(numero)

File: aula11/Fatorial.java
import java.util.Scanner;

public class Fatorial {
	static int fatorial(int n) {
		if (n <= 1) {
			return 1;
		} else {
			return n * fatorial(n-1);
		}
	}
	
	public static void main(String[] args) {
		int numero;
		Scanner console = new Scanner(System.in);
		System.out.print("Digite um número inteiro positivo: ");
		numero = console.nextInt();
		System.out.println(numero+"! = "+fatorial(numero));
	}
}
